src/App.tsx

```diff
@@ -7,33 +13,111 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+interface AppState {
+  hasClock: boolean;
+  clockName: string;
+}
+
+interface TimeState {
+  today: Date;
+}
+
+export class Time extends React.Component<{}, TimeState> {
+  state = {
+    today: new Date(),
+  };
+
+  timerId = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  timerId2 = 0;
```

Comment:
do you need this `timerId2` here? think using just `timerId` in this component should be enough